### ISSUE TITLE
grok svn 1.4 information in twisted.python.versions

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3058]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3058|
|Type|enhancement|
|Created|2008-02-28 15:05:17Z|
|Branch|https://github.com/twisted/twisted/tree/svn14version-3058|

Currently only svn 1.3 internal details are recognized.  1.4 is in widespread use now (and 1.5 is coming pretty soon I guess maybe).  It'd be nice to update this functionality a bit.

Attachments:

* [svn14.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/68f/68ffefb56efac74ed366759a0f5436076ce7f2d9/bc563e51a6cca8a26190786c1bf404e79559c54d.patch) (790 bytes) - added by glyph on 2008-02-28 23:37:43Z - a bad joke about a solution to this issue

<details><summary>Searchable metadata</summary>

```
trac-id__3058 3058
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_svn14version_3058 branches/svn14version-3058
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1204211117000000 1204211117000000
changetime__1204289555000000 1204289555000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [22721]) Branching to 'svn14version-3058'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph removed owner|
|-|-|

A real solution is in the branch.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @glyph|
|-|-|

I tried hard to spot something, but it looks great. Please update the copyright notice of both files, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set status to `closed`|
|-|-|

(In [22725]) Update twisted.python.versions to understand version 8 of SVN "entries" files.

Fixes [#3058](https://github.com/twisted/twisted/issues/3058)

Author: glyph, exarkun

Reviewer: therve

For a long time now, twisted.python.versions has been giving back "rUnknown" for SVN checkouts.  This updates it to understand the more recent format used by svn 1.4, as well as suggesting a very simple framework for dealing with future versions.

This also moves another test module out of twisted.test, continuing the slow process of depopulating that namespace and moving everything to dedicated test packages.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
